analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -147,4 +147,23 @@ public override void ExecutionCompleted()
             ReportIssue(operation, operation.Syntax.ToString());
         }
     }
+
+    private static ProgramState SetConstraintForObjectCreation(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
```

Comment:
The pattern matching makes this very cryptic and hard to follow. Make it `if/else` with one ternary inside instead.

The other comment asks to remove `ConstraintFromArgument` and that will make this way easier to do

Reply:
I took a different approach. Let me know what you think of it.